### ISSUE TITLE
`no-unused-vars` wrong diagnostic when using destructuring and defaults

### ISSUE THREAD

**Issue by lowlighter** [denoland/deno_lint#687]
```ts
const {id, name = id} = {id:"10"} as {[key:string]:string}
console.log(name) //"10"
```
When using destructuring assignement along with default subsequent values, linter report false positives:
```
(no-unused-vars) `id` is never used
const {id, name = id} = {id:"10"} as {[key:string]:string}
       ^^
    at \\test.ts:1:7

    hint: If this is intentional, prefix it with an underscore like `_id`

Found 1 problem
Checked 1 file
```

```
deno 1.9.2 (release, x86_64-pc-windows-msvc)
v8 9.1.269.5
typescript 4.2.2
```

Edit: Actually maybe it is actually expected to do the following?
```ts
const {id:_id, name = _id} = {id:"10"} as {[key:string]:string}
console.log(name) //"10"
```

**Comment by magurotuna** [comment]
Thanks for reporting! Looks like a little bit tricky code that I've never seen before, but should not trigger the lint error.
I've fixed it in #688.
